plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -209,7 +220,7 @@ func (s *SpanReader) readTraceInSpan(ctx context.Context, traceID dbmodel.TraceI
 
 // GetTrace takes a traceID and returns a Trace associated with that traceID
 func (s *SpanReader) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
-	return s.readTrace(ctx, dbmodel.TraceIDFromDomain(traceID))
+	return s.readTrace(ctx, dbmodel.TraceIDFromDomain(traceID), s.tracer)
```

Comment:
you're calling member functions on the same receiver `s`, there is no need to pass it a tracer that is stored in the same object.